packages/lesswrong/components/seasonal/meetupMonth/MeetupMonthBanner.tsx

```diff
@@ -0,0 +1,541 @@
+    }
+    return <LocalEventMapMarkerWrappersInner  onMarkerClick={() => setEverClickedMap(true)} localEvents={events} />
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [acxActive, ifanyoneActive, petrovActive, events])
+  
+  const handleZoomIn = useCallback(() => {
+    setViewport(prev => ({
+      ...prev,
+      zoom: Math.min(prev.zoom + 0.5, 20),
+    }))
+  }, [])
+  
+  const handleZoomOut = useCallback(() => {
+    // Prevent zooming out beyond a safe threshold – when zooming too far out Mapbox starts repeating the
+    // world map horizontally which causes the markers to render offset from their real positions.
+    // We empirically found zoom < 1 to be problematic, so clamp the minimum zoom to 1.
+    setViewport(prev => ({
+      ...prev,
+      zoom: Math.max(prev.zoom - 0.5, 1),
+    }))
+  }, [])
+
+  const handleMeetupTypeClick = useCallback((index: number) => {
+    if (index === currentCarouselIndex) return
+
+    setIsSettingUp(true)
+    setNextCarouselIndex(index)
+
+    setTimeout(() => {
+      setIsSettingUp(false)
+      setIsTransitioning(true)
+      setTimeout(() => {
+        setIsTransitioning(false)
+        setCurrentCarouselIndex(index)
+        setNextCarouselIndex(null)
+      }, 300)
+    }, 10)
+  }, [currentCarouselIndex])
+
+  // Automatically rotate carousel unless the user has clicked on the map
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      if (everClickedMap) return
+      const nextIndex = (currentCarouselIndex + 1) % carouselSections.length
+      handleMeetupTypeClick(nextIndex)
+    }, 30 * 1000)
+    return () => clearInterval(intervalId)
+  }, [currentCarouselIndex, handleMeetupTypeClick, carouselSections.length, everClickedMap])
+
+  if (isLoading) {
+    return <div className={classes.root}>
+      <div className={classes.mapGradient} />
+      <div className={classes.mapGradientRight} />
+    </div>;
+  }
+
+  return <div className={`${classes.root} ${acxActive ? classes.acxMode : ''} ${ifanyoneActive ? classes.ifanyoneMode : ''} ${petrovActive ? classes.petrovMode : ''}`}>
+    <div className={classes.mapGradient}/>
+    <div className={classes.mapGradientRight} />
+    <div 
+      className={classes.mapContainer} 
+      onClick={() => setEverClickedMap(true)}
+    >
+      <div className={classes.mapButtonsContainer}>
+        <div className={classes.mapButtons} onClick={() => setEverClickedMap(true)}>
+          <MagnifyingGlassMinusIcon className={classes.zoomButton} onClick={handleZoomOut} />
+          <MagnifyingGlassPlusIcon className={classes.zoomButton} onClick={handleZoomIn} />
+        </div>
+      </div>
+      <div className={classes.contentContainer}>
+        <div className={classes.textContainer} onClick={() => setEverClickedMap(true)}>
+          {carouselSections.map((section, index) => {
+            
+            const aboutToTransition = isSettingUp && index === nextCarouselIndex
```

Comment:
If these three states are mutually exclusive, this probably would've been better derived as a `carouselState` union of `type CarouselState = 'aboutToTransition' | 'isTransitioningOut' | 'isTransitioningIn';` and everything downstream of that would have an easier time